core/app/models/spree/promotion.rb

```diff
@@ -194,21 +194,7 @@ def line_item_actionable?(order, line_item)
     end
 
     def used_by?(user, excluded_orders = [])
-      [
-        :adjustments,
-        :line_item_adjustments,
-        :shipment_adjustments
-      ].any? do |adjustment_type|
-        user.orders.complete.joins(adjustment_type).where(
-          spree_adjustments: {
-            source_type: 'Spree::PromotionAction',
-            source_id: actions.map(&:id),
-            eligible: true
-          }
-        ).where.not(
-          id: excluded_orders.map(&:id)
-        ).any?
-      end
+      user.orders.complete.joins(:promotions).where.not(spree_orders: { id: excluded_orders.pluck(:id) }).where(spree_promotions: { id: id }).any?
```

Comment:
```suggestion
      user.orders.complete.joins(:promotions).where.not(spree_orders: { id: excluded_orders.map(&:id) }).where(spree_promotions: { id: id }).any?
```

`excluded_orders` can be an array right? it's not AR collection?

Reply:
So far `excluded_orders` is an array of one object, not AR collection: https://github.com/spree/spree/blob/4cc2c2967352d3fe7652bd873e492c074ffcc3da/core/app/models/spree/promotion/rules/one_use_per_user.rb#L11

Reply:
@KacperMekarski so we need to change this to map because array object doesn't have `pluck` method

Reply:
@damianlegawiec done.